engine/config/locales/cy.yml

```diff
@@ -44,3 +44,64 @@ cy:
       close_label: Cau
       descriptive_label_hide: cau’r adran hon
       descriptive_label_show: dangos yr adran hon
+
+  # The rails-i18n gem historically provided these but
+  # the cy locale was removed in 7.0.6 due to lack of support.
+  # See https://github.com/svenfuchs/rails-i18n/pull/1007
+  #
+  # As we only need a small set of translations for date
+  # formatted provided by design system components we bundle
+  # the necessary translations ourselves.
+  date:
+    abbr_day_names:
+      - Sul
+      - Llun
+      - Maw
+      - Mer
+      - Iau
+      - Gwe
+      - Sad
+    abbr_month_names:
+      -
+      - Ion
+      - Chw
+      - Maw
+      - Ebr
+      - Mai
+      - Meh
+      - Gor
+      - Awst
+      - Med
+      - Hyd
+      - Tach
+      - Rha
+    day_names:
+      - Dydd Sul
+      - Dydd Llun
+      - Dydd Mawrth
+      - Dydd Mercher
+      - Dydd Iau
+      - Dydd Gwener
+      - Dydd Sadwrn
+    formats:
+      default: '%d-%m-%Y'
+      long: '%d %B %Y'
+      short: '%d %b'
+    month_names:
+      -
```

Comment:
?

Reply:
If I understand the original code correctly it's because of non-zero-indexing (i.e. January is month 1)